StringSlice

YahorChaika opened the pull request:
### Description
## Ticket reference

### For a given string slice which can have multiple repeated values create set of unique values using map and remove all repeated values from slice respecting the original order.

I wrote a Go program that removes duplicate elements from a string slice while keeping the original order of elements.
The main idea is to use a map to store information about which strings have already been seen, and append only unique values into a new slice.
How the Program Works:
1.A slice of strings is defined, where some values are repeated.
2.An empty slice "result" is created to store only unique strings.
3.A map seen (map[string]bool) is created to mark all strings that have already appeared.
4.The program loops through each element in the original slice:
If the string hasn’t been seen yet, it’s added to "result".
The string is then marked as seen.
5.As a result, we get a new slice containing only unique values.

YahorChaika committed:
StringSlice

YahorChaika commented:
I wrote a Go program that removes duplicate elements from a string slice while keeping the original order of elements.
The main idea is to use a map to store information about which strings have already been seen, and append only unique values into a new slice.
How the Program Works:
1.A slice of strings is defined, where some values are repeated.
2.An empty slice "result" is created to store only unique strings.
3.A map seen (map[string]bool) is created to mark all strings that have already appeared.
4.The program loops through each element in the original slice:
  If the string hasn’t been seen yet, it’s added to "result".
  The string is then marked as seen.
5.As a result, we get a new slice containing only unique values.

EgorKurochkin commented on StringSlice/main.go:
```diff
@@ -0,0 +1,19 @@
+func main() {
+    slice := []string{"apple", "banana", "apple", "orange", "banana", "kiwi"}
+
+    result := []string{}
```
@YahorChaika can we define result slice with help of make() in-built function?

EgorKurochkin reviewed (approved):
Overall looks good, but make function should be used and also you should be able to explain, what is the difference between defining slice with []string{} and make(). 
Good work, Yahor!

YahorChaika committed:
Defining result Slice with the help of make() function

Changed result slice initialization to use make()
- Now the slice is created with length 0 and capacity = len(slice)
- This avoids extra memory allocations while appending
